app/javascript/components/public_health/query/AdvancedFilter.js

```diff
@@ -391,6 +428,66 @@ class AdvancedFilter extends React.Component {
     this.setState({ activeFilterOptions });
   };
 
+  /**
+   * Change value of a filter statement of type multi
+   * @param {Number} statementIndex - Current overall statement filter index
+   * @param {Number} multiIndex - Index of the statement within the multi filter that needs to be updated
+   * @param {*} value - New value for the statement at the multi index
+   */
+  changeMultiValue = (statementIndex, multiIndex, value) => {
+    const currentMultiFilter = this.state.activeFilterOptions[parseInt(statementIndex)];
+    const newValue = [...currentMultiFilter.value];
+    newValue[parseInt(multiIndex)] = value;
+    this.changeValue(statementIndex, newValue);
+  };
+
+  /**
+   * Determines if a field within a multi statement should be disabled
+   * A field is disabled if it already exists in a different multi statement, but not the one it is selected in
+   * @param {String} name - Field name attribute of the dropdown option
+   * @param {Number} statementIndex - Current overall statement filter index
+   * @param {Number} multiIndex - Index of the statement within the multi filter
+   */
+  multiFieldDisabled = (name, multiIndex, statementIndex) => {
+    const currentMultiFilter = this.state.activeFilterOptions[parseInt(statementIndex)];
+    let disabled = false;
+    currentMultiFilter.value.forEach((value, index) => {
```

Comment:
This is totally fine, but couldnt it also be written as:
```
let disabled = currentMultiFilter.value.some((value, index) => value.name === name && index !== multiIndex)

Reply:
fixed here: https://github.com/SaraAlert/SaraAlert/pull/821/commits/0beb57efa303876fa3aea5c08ba570a5967b80ce